src/v/cluster/partition.h

```diff
@@ -151,7 +151,11 @@ class partition {
 
     ss::future<std::error_code>
     transfer_leadership(std::optional<model::node_id> target) {
-        return _raft->do_transfer_leadership(target);
+        if (_rm_stm) {
+            return _rm_stm->transfer_leadership(target);
```

Comment:
I have a general question, can the livelock happen when leader is elected because hearbeats were lost ?